soc/xtensa/intel_adsp/common/boot.c

```diff
@@ -16,6 +16,11 @@
 #include <cpu_init.h>
 #include "manifest.h"
 
+#if CONFIG_ADSP_DYNAMIC_CLOCK_SWITCHING
```

Comment:
Nitpick: #if'ing headers is usually pointless.  C headers are cheap to parse, and with very few exceptions all Zephyr headers should be includable in any context regardless of build-time configuration.

Reply:
.h files is one of C/C++ worst headaches:
- https://github.com/zephyrproject-rtos/zephyr/issues/41543

>  all Zephyr headers should be includable in any context regardless of build-time configuration.

Indeed, one way to help maintain this property is to parse headers even when they're not in use because this detects conflicts and other problems SOONER, not much later when someone turns on some rarely use CONFIG_

Reply:
This feature is not enabled on CAVS, and this header needs to be under #ifdef in this common file.

Reply:
I just tried and solved this problem by adding an empty `soc/xtensa/intel_adsp/cavs/include/pm_data.h` placeholder.

Simpler, less unusual and more consistent. Now `ace/include/pm_data.h` is parsed every time which provides more coverage and can find more conflicts.

You just need to add at least one line to pass CI checks: good opportunity to mention the purpose of the placeholder.

Reply:
@marc-hb thanks for the hint.